src/main/java/org/myrobotlab/service/InMoov2.java

```diff
@@ -880,21 +875,21 @@ public void moveHead(Integer neck, Integer rothead, Integer rollNeck) {
     moveHead((double) neck, (double) rothead, null, null, null, (double) rollNeck);
   }
 
-  public void moveHeadBlocking(Double neck, Double rothead) {
+  public void moveHeadBlocking(Double neck, Double rothead) throws InterruptedException, TimeoutException {
     moveHeadBlocking(neck, rothead, null);
   }
 
-  public void moveHeadBlocking(Double neck, Double rothead, Double rollNeck) {
+  public void moveHeadBlocking(Double neck, Double rothead, Double rollNeck) throws InterruptedException, TimeoutException {
     moveHeadBlocking(neck, rothead, null, null, null, rollNeck);
   }
 
-  public void moveHeadBlocking(Double neck, Double rothead, Double eyeX, Double eyeY, Double jaw) {
+  public void moveHeadBlocking(Double neck, Double rothead, Double eyeX, Double eyeY, Double jaw) throws InterruptedException, TimeoutException {
     moveHeadBlocking(neck, rothead, eyeX, eyeY, jaw, null);
   }
 
-  public void moveHeadBlocking(Double neck, Double rothead, Double eyeX, Double eyeY, Double jaw, Double rollNeck) {
-    // the "right" way
-    sendToPeer("head", "moveToBlocking", neck, rothead, eyeX, eyeY, jaw, rollNeck);
+  public void moveHeadBlocking(Double neck, Double rothead, Double eyeX, Double eyeY, Double jaw, Double rollNeck) throws InterruptedException, TimeoutException {
+    sendBlocking(getPeerName("head"), "moveToBlocking", neck, rothead, eyeX, eyeY, jaw, rollNeck);
```

Comment:
Bubbling up the checked exceptions is probably not needed; the try/catch can probably be done here at a minimum if not in sendBlocking itself

Reply:
the library method sendBlocking always need to throw on timeout, but I agree this stuff doesn't